src/cluster.c

```diff
@@ -1064,7 +1068,7 @@ ShardGroup **ShardGroupsParse(RedisModuleCtx *ctx,
 fail:
     *len = 0;
 
-    for (int j = 0; j <= i; j++) {
+    for (unsigned int j = 0; j < created_shard_count; j++) {
```

Comment:
```suggestion
    for (int j = 0; j < created_shard_count; j++) {
```
I think linter will complain about this

Reply:
hmm, I mean for the created_shard_count to be unsigned.  will fix it there